pkg/build/apis/build/validation/validation.go

```diff
@@ -699,7 +699,13 @@ func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorLis
 
 func validatePostCommit(spec buildapi.BuildPostCommitSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if spec.Script != "" && len(spec.Command) > 0 {
+	if len(spec.Command) == 0 && len(spec.Args) != 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec, "args must be used with command"))
+	}
+	if len(spec.Script) != 0 && len(spec.Args) != 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec, "args cannot be used with script"))
```

Comment:
why not?  the code seems to support it (appends the args to the script)

Reply:
Currently you get this error when trying to specify args with --script

```
$ oc set build-hook bc/mybuild --post-commit --script="/var/lib/test-image.sh" -- arg1 arg2
error: a command cannot be specified when using the --script argument
```
which comes from https://github.com/openshift/origin/blob/master/pkg/oc/cli/set/buildhook.go#L176-L178 which is probably incorrect behavior, fixing.

Reply:
fixed in current push